www/js/app.js

```diff
@@ -1300,8 +1317,75 @@ function displayFileSelect () {
+            } else {
+                if (webKitFileList === null) {
+                    const element = localStorage.getItem('zimFilenames').split('|').length === 1 ? 'archiveFiles' : 'folderSelect';
+                    // console.log(localStorage.getItem('zimFilenames').split('|'));
+                    document.getElementById(element).click();
+                    return;
+                }
+                const files = abstractFilesystemAccess.getSelectedZimFromWebkitList(webKitFileList, e.target.value)
+                setLocalArchiveFromFileList(files);
+            }
+        });
+    }
+    if (params.isFileSystemApiSupported && !isFireFoxOsNativeFileApiAvailable) {
+        // Handles Folder selection when showDirectoryPicker is supported
+        document.getElementById('folderSelect').addEventListener('click', async function (e) {
+            e.preventDefault();
+            const previousZimFiles = await abstractFilesystemAccess.selectDirectoryFromPickerViaFileSystemApi()
+            if (previousZimFiles.length !== 0) setLocalArchiveFromFileList(previousZimFiles);
+        })
+    }
+    if (params.isWebkitDirApiSupported && !params.isFileSystemApiSupported && !isFireFoxOsNativeFileApiAvailable) {
+        // Handles Folder selection when webkitdirectory is supported but showDirectoryPicker is not
+        document.getElementById('folderSelect').addEventListener('change', async function (e) {
+            e.preventDefault();
+            const filenames = [];
+
+            const previousZimFile = []
+            const lastFilename = localStorage.getItem('previousZimFileName') ?? '';
+            const filenameWithoutExtension = lastFilename.replace(/\.zim\w\w$/i, '');
+            const regex = new RegExp(`\\${filenameWithoutExtension}.zim\\w\\w$`, 'i');
+
+            for (const file of e.target.files) {
+                filenames.push(file.name);
+                if (regex.test(file.name) || file.name === lastFilename) previousZimFile.push(file);
+            }
+            webKitFileList = e.target.files;
+            localStorage.setItem('zimFilenames', filenames.join('|'));
+            // will load the old file if the selected folder contains the same file
+            if (previousZimFile.length !== 0) setLocalArchiveFromFileList(previousZimFile);
+            await abstractFilesystemAccess.updateZimDropdownOptions(filenames, previousZimFile.length !== 0 ? lastFilename : '');
+        })
+    }
+    if (params.isFileSystemApiSupported && !isFireFoxOsNativeFileApiAvailable) {
+        // Handles File selection when showOpenFilePicker is supported and uses the filesystem api
+        document.getElementById('archiveFiles').addEventListener('click', async function (e) {
+            e.preventDefault();
+            const files = await abstractFilesystemAccess.selectFileFromPickerViaFileSystemApi(e);
+            setLocalArchiveFromFileList(files);
+        });
+    } else {
```

Comment:
So this would become:
```
} else {
    useLegacyFilePicker();
}
```
And you then put the code below in a function of its own (outside of this function scope), above `handleGlobalDragover` below.